### ISSUE TITLE
Requested RAM to run a complete training cycle

### ISSUE THREAD

**Issue by GiuseppeT00** [LangeMoritz/GNN_decoder#1]
Hi there, I'm trying to reproduce the training cycle corresponding to Figure 4 of the paper, for which, as I understood, parameters are the following: {code size: 5, repetitions: 5, epochs: 300, batch size: 1000, lr: 0.0001, decreased at 0.00001 at epoch 250, buffer size: 1000, replacements per iteration: 0 & 250, test size: 10000, error rates: [0.001, 0.002, 0.003, 0.004, 0.005]}, where batch size (it's value has been denoted as 1000 on page 10, Appendix A of the paper) and buffer size are chosen in such a way that we have 5mln training samples (1mln for each error rate), while test size is chosen such that we have 50000 testing samples (10000 for each error rate).
Assuming that these are right parameters to reproduce that experiment, how much RAM memory did you use for that purpose? 
I tried with 256gb of RAM but I got an OOM error by slurm, which refers to cpu ram, not gpu one (I used an A100 80gb, so it's sufficient).

Thank you in advance for your time.

**Comment by LangeMoritz** [comment]
Hi! Thanks for your interest and your question. The parameters for Figure 4 are indeed the ones you described. Do you train on GPU or CPU? We trained on a A100fat GPU with 256GB of system memory.

**Comment by GiuseppeT00** [comment]
Hi! I want firstly thank you for your time, I solved the memory problem. I have other two questions:
- When you talk about logical failure rate in Figure 7, do you refer to it as ``logical_failure_rate = 1 - logical_accuracy``?
- Remaining on Figure 7, (so referring to "IV.3. Surface code with perfect stabilizers"), can you confirm that the same GNN was trained (same number of training samples, learning rate and epochs - 600? - as before), by only changing error rates p (which become ``p=[0.01, 0.05, 0.10, 0.15]``) and with code distances ``d=[3, 5, 7, 9, 11, 13, 15]`` for both GNN & MWPM? If so, what did you set as ``d_t`` parameter? Is it always ``d_t = 1`` or ``d_t = d`` for that experiment, independent from d value? 

Thank you again for your availability!

**Comment by LangeMoritz** [comment]
Sure, you are welcome. 

- indeed, the logical failure rate is defined as 1 - logical accuracy. 

- correct, same network architectures, same hyper-parameters, but new mix of error rates as you mentioned. The number of training epochs increases from 100 (for d=3) to 1100 (for d = 15). For perfect stabilizer measurements, d_t = 1 for all d.

**Comment by GiuseppeT00** [comment]
Perfect, I really thank you for your time, and I also want to congratulate for the paper. It's really a good idea, with good results :).
Hope you the best!

**Comment by GiuseppeT00** [comment]
Perfect, I really thank you for your time, and I also want to congratulate for the paper. It's really a good idea, with good results :).
Wish you the best!
